lib/crypto.js

```diff
@@ -0,0 +1,166 @@
+/*!
+ * Copyright (c) 2020 Digital Bazaar, Inc. All rights reserved.
+ */
+
+import {
+  sign,
+  verify,
+  createPrivateKey,
+  createPublicKey
+} from 'crypto';
+
+// used to export node's public keys to buffers
+const publicKeyEncoding = {format: 'der', type: 'spki'};
+// used to export node's private keys to buffers
+const privateKeyEncoding = {format: 'der', type: 'pkcs8'};
+// used to turn private key bytes into a buffer in DER format
+const DER_PRIVATE_KEY_PREFIX = Buffer.from(
+  '302e020100300506032b657004220420', 'hex');
+// used to turn public key bytes into a buffer in DER format
+const DER_PUBLIC_KEY_PREFIX = Buffer.from('302a300506032b6570032100', 'hex');
+
+/**
+ * The key material is the part of the buffer after the DER Prefix.
+ *
+ * @param {Buffer} buffer - A DER encoded key buffer.
+ *
+ * @throws {Error} If the buffer does not contain a valid DER Prefix.
+ *
+ * @returns {Buffer} The key material part of the Buffer.
+*/
+function getKeyMaterial(buffer) {
+  if(buffer.indexOf(DER_PUBLIC_KEY_PREFIX, 0) === 0) {
```

Comment:
```suggestion
  if(buffer.indexOf(DER_PUBLIC_KEY_PREFIX) === 0) {
```

Reply:
while unlikely, I think it is better to start at index 0 to ensure that the public key material after the prefix is not somehow a match for the prefix itself.